error occurred:multiple definition of `mlpack::rl::ContinuousActionEnv::Action conflicts.When the header file is referenced by multiple files, the reinforcement learning library conflicts

I'm new to mlpack. I want to do some simple experiments with reinforcement learning, but there are some problems.
I found that when the header file is referenced individually, it can be linked normally, but when multiple files are referenced, the links will conflict, as follows:
```
Scanning dependencies of target test
[ 33%] Building CXX object CMakeFiles/test.dir/main.cpp.o
[ 66%] Building CXX object CMakeFiles/test.dir/my_dqn.cpp.o
[100%] Linking CXX executable test
CMakeFiles/test.dir/my_dqn.cpp.o:(.bss+0x0): multiple definition of `mlpack::rl::ContinuousActionEnv::Action::size'
CMakeFiles/test.dir/main.cpp.o:(.bss+0x0): first defined here
CMakeFiles/test.dir/my_dqn.cpp.o:(.bss+0x8): multiple definition of `mlpack::rl::ContinuousActionEnv::State::dimension'
CMakeFiles/test.dir/main.cpp.o:(.bss+0x8): first defined here
CMakeFiles/test.dir/my_dqn.cpp.o:(.bss+0x10): multiple definition of `mlpack::rl::DiscreteActionEnv::Action::size'
CMakeFiles/test.dir/main.cpp.o:(.bss+0x10): first defined here
CMakeFiles/test.dir/my_dqn.cpp.o:(.bss+0x18): multiple definition of `mlpack::rl::DiscreteActionEnv::State::dimension'
CMakeFiles/test.dir/main.cpp.o:(.bss+0x18): first defined here
collect2: error: ld returned 1 exit status
CMakeFiles/test.dir/build.make:117: recipe for target 'test' failed
make[2]: *** [test] Error 1
CMakeFiles/Makefile2:94: recipe for target 'CMakeFiles/test.dir/all' failed
make[1]: *** [CMakeFiles/test.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make: *** [all] Error 2
```
The project is like this：
CMakeLists.txt
```
cmake_minimum_required(VERSION 3.7.0)
project(test)
SET(CMAKE_CXX_FLAGS "-std=c++11 -O3 -fopenmp")
set(CMAKE_INCLUDE_CURRENT_DIR ON)
add_compile_options(-fext-numeric-literals)
include_directories(mlpack)
add_executable(test main.cpp my_dqn.cpp)
target_link_libraries(test mlpack)
```
main.cpp
```
#include "my_dqn.h"

int main(int argc, char *argv[])
{
    return 0;
}
```
my_dqn.h
`#include <mlpack/methods/reinforcement_learning/environment/env_type.hpp>`
my_dqn.cpp
`#include <my_dqn.h>`
I don't know how to correct it.Thank you for your help~

--- a/src/mlpack/methods/reinforcement_learning/environment/CMakeLists.txt
+++ b/src/mlpack/methods/reinforcement_learning/environment/CMakeLists.txt
@@ -2,6 +2,7 @@
 # Anything not in this list will not be compiled into mlpack.
 set(SOURCES
   env_type.hpp
+  env_type.cpp
   mountain_car.hpp
   cart_pole.hpp
   continuous_mountain_car.hpp

--- a/src/mlpack/methods/reinforcement_learning/environment/env_type.cpp
+++ b/src/mlpack/methods/reinforcement_learning/environment/env_type.cpp
@@ -1,0 +1,27 @@
+/**
+ * @file methods/reinforcement_learning/environment/env_type.cpp
+ * @author Nishant Kumar
+ *
+ * This file defines the static variables used by the discrete and continuous
+ * environments.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "env_type.hpp"
+
+namespace mlpack {
+namespace rl {
+
+// Instantiate static members.
+
+size_t DiscreteActionEnv::State::dimension = 0;
+size_t DiscreteActionEnv::Action::size = 0;
+
+size_t ContinuousActionEnv::State::dimension = 0;
+size_t ContinuousActionEnv::Action::size = 0;
+
+} // namespace rl
+} // namespace mlpack

--- a/src/mlpack/methods/reinforcement_learning/environment/env_type.hpp
+++ b/src/mlpack/methods/reinforcement_learning/environment/env_type.hpp
@@ -105,8 +105,6 @@ class DiscreteActionEnv
    */
   bool IsTerminal(const State& /* state */) const { return false; }
 };
-size_t DiscreteActionEnv::State::dimension = 0;
-size_t DiscreteActionEnv::Action::size = 0;
 
 /**
  * To use the dummy environment, one may start by specifying the state and
@@ -201,8 +199,6 @@ class ContinuousActionEnv
    */
   bool IsTerminal(const State& /* state */) const { return false; }
 };
-size_t ContinuousActionEnv::State::dimension = 0;
-size_t ContinuousActionEnv::Action::size = 0;
 
 } // namespace rl
 } // namespace mlpack

